src/providers/twitch/api/Helix.cpp

```diff
@@ -2631,6 +2631,96 @@ void Helix::updateShieldMode(
+{
+    using Error = HelixSendShoutOutError;
+
+    QUrlQuery urlQuery;
+    urlQuery.addQueryItem("from_broadcaster_id", fromBroadcasterID);
+    urlQuery.addQueryItem("to_broadcaster_id", toBroadcasterID);
+    urlQuery.addQueryItem("moderator_id", moderatorID);
+
+    this->makeRequest("chat/shoutouts", urlQuery)
+        .type(NetworkRequestType::Post)
+        .header("Content-Type", "application/json")
+        .onSuccess([successCallback](auto result) -> Outcome {
+            if (result.status() != 204)
+            {
+                qCWarning(chatterinoTwitch)
+                    << "Success result for sending shoutout was "
+                    << result.status() << "but we expected it to be 204";
+            }
+
+            successCallback();
+            return Success;
+        })
+        .onError([failureCallback](auto result) -> void {
+            const auto obj = result.parseJson();
+            auto message = obj["message"].toString();
+
+            switch (result.status())
+            {
+                case 400: {
+                    if (message.startsWith("The broadcaster may not give "
+                                           "themselves a Shoutout.",
+                                           Qt::CaseInsensitive))
+                    {
+                        failureCallback(Error::UserIsBroadcaster, message);
+                    }
+                    else if (message.startsWith(
+                                 "The broadcaster is not streaming live or "
+                                 "does not have one or more viewers.",
+                                 Qt::CaseInsensitive))
+                    {
+                        failureCallback(Error::BroadcasterNotLive, message);
+                    }
+                    else
+                    {
+                        failureCallback(Error::UserNotAuthorized, message);
+                    }
+                    break;
+                }
+                case 401: {
+                    if (message.startsWith("Missing scope",
+                                           Qt::CaseInsensitive))
+                    {
+                        failureCallback(Error::UserMissingScope, message);
+                    }
+                    else
+                    {
+                        failureCallback(Error::UserNotAuthorized, message);
+                    }
+                    break;
+                }
+                case 403: {
+                    failureCallback(Error::UserNotAuthorized, message);
+                    break;
+                }
+                case 429: {
+                    failureCallback(Error::Ratelimited, message);
+                    break;
+                }
+                    // Helix returns 500 when user is not mod
+                case 500: {
+                    failureCallback(Error::UserNotMod,
+                                    "You are not a moderator.");
+                    break;
```

Comment:
I would prefer to treat 500s as unhandled errors - just forward the message we get from twitch

Reply:
Twitch likes to give an empty message on 500s so there needs to be a genetic "twitch fucked up" error case and one that forwards a non-empty message.